### ISSUE TITLE
Android (Non Patrol) Release Build - java.lang.AbstractMethodError: abstract method "java.lang.Object java.lang.ClassValue.computeValue(java.lang.Class)

### ISSUE THREAD

**Issue by mattbajorek** [leancodepl/patrol#2033]
### Steps to reproduce

`flutter run --release --verbose`

### Actual results

App (not patrol) release build on Android immediate crashes after start. Temporary solution I found was to uninstall patrol from pubspec.yaml before the release build. However, it would be good to figure out the underlying problem. It seems like there is a package conflict. I am able to build the example app in release mode and the example app with the same Flutter packages in pubspec.yaml as my app. Any ideas on what could be causing this?

### Logs

[logs.txt](https://github.com/leancodepl/patrol/files/13780769/logs.txt)

### Patrol version

patrol: ^3.3.0

### Patrol Doctor output

<details>
<summary>Patrol Doctor output</summary>

```
Patrol CLI version: 2.5.0
Program adb found in /Users/matthewbajorek/Library/Android/sdk/platform-tools/adb
Env var $ANDROID_HOME set to /Users/matthewbajorek/Library/Android/sdk
Program xcodebuild found in /usr/bin/xcodebuild
Program ideviceinstaller found in /usr/local/bin/ideviceinstaller
```

</details>

### Flutter Doctor output

<details>
<summary>Flutter Doctor output</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.16.3, on macOS 14.1.2 23B92 darwin-x64, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 15.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.3)
[✓] VS Code (version 1.85.1)
[✓] Connected device (3 available)
[✓] Network resources

• No issues found!
```

</details>

**Comment by piotruela** [comment]
Hi @mattbajorek. Sory for the late reply. Is the problem still there? Were you able to reproduce the issue in the example flutter project?

**Comment by VSAlena** [comment]
> Hi @mattbajorek. Sory for the late reply. Is the problem still there? Were you able to reproduce the issue in the example flutter project?


Yes, the issue with the build crash in release mode right after launch still exists.
In my project, I'm using the following versions:

patrol: 3.4.0
patrol cli: 2.6.0
flutter: 3.16.3
There was no such error in patrol version 2.2.5, but we had to change the patrol version because we updated the flutter version.

Please, fix this issue so that we can continue using patrol for integration tests.

Thank you!

**Comment by pdenert** [comment]
Hey, @mattbajorek are you still interested in resolving this issue?

**Comment by pdenert** [comment]
Hey, @VSAlena are you still interested in resolving this issue?
